jaeger-core/src/main/java/com/uber/jaeger/utils/RateLimiter.java

```diff
@@ -22,20 +22,20 @@
 package com.uber.jaeger.utils;
 
 public class RateLimiter {
-  private final double creditsPerSecond;
   private final double creditsPerNanosecond;
   private final Clock clock;
   private double balance;
+  private double maxBalance;
   private long lastTick;
 
-  public RateLimiter(double creditsPerSecond) {
-    this(creditsPerSecond, new SystemClock());
+  public RateLimiter(double creditsPerSecond, double maxBalance) {
+    this(creditsPerSecond, maxBalance, new SystemClock());
   }
 
-  public RateLimiter(double creditsPerSecond, Clock clock) {
+  public RateLimiter(double creditsPerSecond, double maxBalance, Clock clock) {
```

Comment:
this rate limiter exposes the notion of item cost to the user (and works with variable item costs), so I don't think it's unreasonable to ask people for the other two params which are related to the "balance" maintained by the algorithm.